pandas/_libs/tslibs/period.pyx

```diff
@@ -1268,6 +1268,54 @@ cdef str _period_strftime(int64_t value, int freq, bytes fmt):
     return result
 
 
+def period_array_strftime(
+    ndarray values, int dtype_code, object na_rep, str date_format
+):
+    """
+    Vectorized Period.strftime used for PeriodArray._format_native_types.
+
+    Parameters
+    ----------
+    values : ndarray[int64_t], ndim unrestricted
+    dtype_code : int
+        Corresponds to PeriodDtype._dtype_code
+    na_rep : any
+    date_format : str or None
+    """
+    cdef:
+        Py_ssize_t i, n = values.size
+        int64_t ordinal
+        object item_repr
```

Comment:
This can't be a `str`?

Reply:
In _format_native_types it is annotated as `str | float`

Reply:
woops, thought you were referring to na_rep.  this can't be str bc na_rep isn't str

Reply:
Darn, (odd na_rep can be a float but okay)